### ISSUE TITLE
aria-hidden with focusable content: Expectation is ambiguous

### ISSUE THREAD

**Issue by kasperisager** [auto-wcag/auto-wcag#315]
The expectation for https://auto-wcag.github.io/auto-wcag/rules/SC4-1-2-aria-hidden-focus.html is currently ambiguous as the term _contain_ is never defined. For example, given the following snippet of HTML...

```html
<div aria-hidden="true">
  <button>I'm focusable</button>
</div>

<script>
  {
    const host = document.querySelector("div");
    const root = host.attachShadow({ mode: "open" });

    root.innerHTML = `
      <p>I'm not focusable</p>
    `;
  }
</script>
```

...which elements do the `<div>` _contain_? In well-defined terms, the `<div>` has the following:

- A single [descendant](https://www.w3.org/TR/dom41/#concept-tree-descendant): `<button>I'm focusable</button>`.
- Two [shadow-including descendants](https://www.w3.org/TR/dom41/#concept-shadow-including-descendant): `<button>I'm focusable</button>` and `<p>I'm not focusable</p>`.
- A single [descendant](https://www.w3.org/TR/dom41/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree): `<p>I'm not focusable</p>`.

If we interpret _contain_ to mean any of the first two options, the snippet will fail the rule, while the third option would actually pass the rule.

**Comment by annethyme** [comment]
@WilcoFiers, could we maybe take this up on the f2f next week?

**Comment by annethyme** [comment]
@kasperisager, can Siteimprove come up with a proposal?

**Comment by EmmaJP** [comment]
So is the issue that a web page is not necessarily a static thing? And that the DOM could change over time? Do ACT Rules need to cover that in some way, or is that up to the tool makers?

Or is the issue that DOM tree, shadow DOM and accessibility tree can contain different things at the same time?

**Comment by annethyme** [comment]
Auto-WCAG conclusion: Go ahead with the definition, but make sure it's readable and not too techy.

**Comment by annethyme** [comment]
Will be handled by https://github.com/auto-wcag/auto-wcag/pull/432
